webrender/src/device/gfx/render_pass.rs

```diff
@@ -35,4 +36,134 @@ impl<B: hal::Backend> RenderPass<B> {
+        device: &B::Device,
+        surface_format: hal::format::Format,
+        depth_format: hal::format::Format,
+    ) -> HalRenderPasses<B> {
+        let attachment_r8 = hal::pass::Attachment {
+            format: Some(hal::format::Format::R8Unorm),
+            samples: 1,
+            ops: hal::pass::AttachmentOps::new(
+                hal::pass::AttachmentLoadOp::DontCare,
+                hal::pass::AttachmentStoreOp::Store,
+            ),
+            stencil_ops: hal::pass::AttachmentOps::DONT_CARE,
+            layouts: hal::image::Layout::ColorAttachmentOptimal
+                .. hal::image::Layout::ColorAttachmentOptimal,
+        };
+
+        let attachment_bgra8 = hal::pass::Attachment {
+            format: Some(surface_format),
+            samples: 1,
+            ops: hal::pass::AttachmentOps::new(
+                hal::pass::AttachmentLoadOp::DontCare,
+                hal::pass::AttachmentStoreOp::Store,
+            ),
+            stencil_ops: hal::pass::AttachmentOps::DONT_CARE,
+            layouts: hal::image::Layout::ColorAttachmentOptimal
+                .. hal::image::Layout::ColorAttachmentOptimal,
+        };
+
+        let attachment_depth = hal::pass::Attachment {
+            format: Some(depth_format),
+            samples: 1,
+            ops: hal::pass::AttachmentOps::new(
+                hal::pass::AttachmentLoadOp::DontCare,
+                hal::pass::AttachmentStoreOp::Store,
+            ),
+            stencil_ops: hal::pass::AttachmentOps::DONT_CARE,
+            layouts: hal::image::Layout::DepthStencilAttachmentOptimal
+                .. hal::image::Layout::DepthStencilAttachmentOptimal,
+        };
+
+        let subpass_r8 = hal::pass::SubpassDesc {
+            colors: &[(0, hal::image::Layout::ColorAttachmentOptimal)],
+            depth_stencil: None,
+            inputs: &[],
+            resolves: &[],
+            preserves: &[],
+        };
+
+        let subpass_depth_r8 = hal::pass::SubpassDesc {
+            colors: &[(0, hal::image::Layout::ColorAttachmentOptimal)],
+            depth_stencil: Some(&(1, hal::image::Layout::DepthStencilAttachmentOptimal)),
+            inputs: &[],
+            resolves: &[],
+            preserves: &[],
+        };
+
+        let subpass_bgra8 = hal::pass::SubpassDesc {
+            colors: &[(0, hal::image::Layout::ColorAttachmentOptimal)],
+            depth_stencil: None,
+            inputs: &[],
+            resolves: &[],
+            preserves: &[],
+        };
+
+        let subpass_depth_bgra8 = hal::pass::SubpassDesc {
+            colors: &[(0, hal::image::Layout::ColorAttachmentOptimal)],
+            depth_stencil: Some(&(1, hal::image::Layout::DepthStencilAttachmentOptimal)),
+            inputs: &[],
+            resolves: &[],
+            preserves: &[],
+        };
+
+        let dependency = hal::pass::SubpassDependency {
+            passes: hal::pass::SubpassRef::Pass(0) .. hal::pass::SubpassRef::Pass(0),
```

Comment:
we don't need dependency between Pass(0) and Pass(0) :)

Reply:
Removed it in https://github.com/szeged/webrender/pull/311/files#diff-cdf03c96bd34cb98c2d09b0be7658232